core/include/emulate.h

```diff
@@ -0,0 +1,44 @@
+#ifndef HAX_CORE_EMULATE_H_
+#define HAX_CORE_EMULATE_H_
+
+#include "../include/hax.h"
+#include "emulate_ops.h"
+
+/* Access completed successfully */
+#define EM_CONTINUE        0
+
+struct em_operand_t {
+    unsigned int bytes;
```

Comment:
nit: Any reason not to use a fixed width integer type like `uint32_t`? I actually don't know what `bytes` refers to here.